### PR TITLE
feat: requesting new integration ui to accept users preferences about the new integrations

### LINKED ISSUE #37854: [Task]: Implement the UI for Request a new Integration
### SubTasks

**UX/UI:**
- Add a fixed bottom bar in the Datasources list page with the text: “Are you looking for something different? Request a new integration.”
- When the button “Request a new integration” is clicked, a modal should open allowing the user to provide: 
   Integration (mandatory)
   Tell us more about your use case (optional)
   Email (optional, pre-filled by Appsmith)
- Ensure the “Request integration” button is disabled if the “Integration” field is empty.
- When the request is submitted, Appsmith shows a success toast message "Thank you! We are looking into your request."
   
[Figma link
](https://www.figma.com/design/kbU9xPv44neCfv1FFo9Ndu/User-Activation?node-id=5998-8874&node-type=section&t=6lW0FVSSmQu5Bx7d-0)

image

**Metrics:**
- REQUEST_INTEGRATION_CTA to check the number of users who clicked the button “Request a new integration”.
- REQUEST_INTEGRATION_SUBMITTED to check the Number of users who clicked “Request integration” in the modal, with properties:
         - integration_name
         - use_case_description
         - email

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
This PR introduces a way to accept user's preferences about adding new types of integrations. 
This feature is behind a feature flag : `ab_request_new_integration_enabled`
This fires two events : `REQUEST_INTEGRATION_CTA`  & `REQUEST_INTEGRATION_SUBMITTED`
Screenshot 2024-12-06 at 3 30 19 PM

Fixes #37854
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Datasource"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12231251196>
> Commit: cb2cf80705959117baba8ac20aee18b4a869ec0d
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12231251196&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Datasource`
> Spec:
> <hr>Mon, 09 Dec 2024 08:42:36 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Release Notes

- **New Features**
  - Introduced a new integration request form, allowing users to submit requests for new integrations.
  - Added a modal for requesting new integrations with appropriate labels and error messages.
  - Enhanced input handling to support textarea fields in forms.
  - Implemented a feature flag to conditionally enable the integration request functionality.
  - Added new event tracking for integration request actions to enhance user engagement insights.
  - Added constants and messages to improve user guidance during the integration request process.

- **Bug Fixes**
  - Improved error message visibility for form fields based on user interaction.

- **Documentation**
  - Updated interface and component declarations to reflect new features and enhancements.

### CODE REVIEW

File: app/client/src/pages/Editor/IntegrationEditor/RequestNewIntegration/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      <ModalTrigger>{children}</ModalTrigger>
+      <ModalContentWrapper>
+        <ModalHeader>
+          {createMessage(REQUEST_NEW_INTEGRATIONS.REQUEST_MODAL_HEADING)}
+        </ModalHeader>
+        <Form closeModal={() => setOpen(false)} />
+      </ModalContentWrapper>
+    </Modal>
+  );
+}
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/IntegrationEditor/RequestNewIntegration/index.tsx:40]: Not required

File: app/client/src/pages/Editor/IntegrationEditor/RequestNewIntegration/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <RequestNewIntegrationWrapper gap="spaces-5">
+      <p>{createMessage(REQUEST_NEW_INTEGRATIONS.UNABLE_TO_FIND)}</p>
+      <RequestModal>
+        <Button kind="secondary" onClick={onRequestButtonClick}>
+          {createMessage(REQUEST_NEW_INTEGRATIONS.REQUEST_BUTTON)}
+        </Button>
+      </RequestModal>
+    </RequestNewIntegrationWrapper>
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/IntegrationEditor/RequestNewIntegration/index.tsx:61]: Not required!

File: app/client/src/components/utils/ReduxFormTextField.tsx [hunk 3]
```diff
@@ -30,14 +30,16 @@ const renderComponent = (
   ) : (
     <Input
       {...componentProps.input}
-      {...componentProps}
+      {...(componentProps.type === "textarea"
+        ? omit(componentProps, "type")
```
Comment (sneha122) [inline app/client/src/components/utils/ReduxFormTextField.tsx:34]: Why are we omitting type here when type is textarea? Can we add appropriate comment here?

**Review by sneha122 — DISMISSED** [review]
LGTM, Added a nit comment to add code comments

File: app/client/src/components/utils/ReduxFormTextField.tsx (hunk 3 above)
Comment (AmanAgarwal041) [reply app/client/src/components/utils/ReduxFormTextField.tsx:34]: Sure, will add this.

**Review by sneha122 — APPROVED** [review]

**Comment by AmanAgarwal041** [comment]
Merged this PR. Will update any new design changes later cc @carinanfonseca

**Pull request merged**